lib/api/controller/document.js

```diff
@@ -295,27 +292,26 @@ class DocumentController extends NativeController {
     const source = this.getBoolean(request, 'source');
     const { index, collection } = this.getIndexAndCollection(request);
 
-    return this.kuzzle.validation.validate(request, false)
-      .then(newRequest => {
-        modifiedRequest = newRequest;
+    const modifiedRequest = await this.kuzzle.validation.validate(request, false);
 
-        return this.publicStorage.update(
-          index,
-          collection,
-          id,
-          content,
-          { refresh, retryOnConflict, userId });
-      })
-      .then(_response => {
-        response = _response;
-        return this.kuzzle.notifier.notifyDocumentUpdate(modifiedRequest, response);
-      })
-      .then(() => {
-        if (!source) {
-          return {_id: response._id, _version: response._version};
-        }
-        return response;
-      });
+    const updatedDocument = await this.publicStorage.update(
+      index,
+      collection,
+      id,
+      content,
+      { refresh, retryOnConflict, userId });
+
+    await this.kuzzle.notifier.notifyDocumentUpdate(modifiedRequest, updatedDocument);
+
+    if (source) {
+      return updatedDocument;
+    }
+
+    return {
+      _id: updatedDocument._id,
+      _source: content,
+      _version: updatedDocument._version,
+    };
```

Comment:
```suggestion
   return source
      ? updatedDocument
      : {
      _id: updatedDocument._id,
      _source: content,
      _version: updatedDocument._version,
    };
```

Reply:
The ternary operator is only useful when it doesn't make the code less readable, and this doesn't appear to be the case here.

Reply:
well I wont fight for this, the only thing I see is a gain of space, not less or more readable code

Reply:
Gaining space doesn't really matter, as it may or may not have an impact on maintenability.
When 2 syntaxes are strictly equivalent, the most important factor of choice should always be "what's the easiest to read?"

The fact that the indentation of the returned object is off hints that using a ternary here might not be a good choice.